File: API/Entity/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entity
{
    public class Product
    {
        [Required]
        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int InsertedBy { get; set; }
        public DateTime InsertedOn { get; set; }

        public int CategoryId { get; set; }
        public float Price { get; set; }
        public bool IsSale { get; set; }

        [ForeignKey("InsertedBy")]
        public User User { get; set; }

        [ForeignKey("CategoryId")]
        public Category Category { get; set; }

        public ICollection<UserFavorites> UserFavorites { get; set; }
        public ICollection<Conversation> Conversations { get; set; }
    }
}

File: client/src/app/_Models/Product.ts
import { Photo } from "./Photo";

export interface Product {
  productId: number;
  name: string;
  description: string;
  insertedBy: number;
  insertedOn: Date;
  categoryId: number;
  price: number;
  isSale: boolean;
  owner: string;
  address: string;
  categoryName: string;
  photos: Photo[]
}